simulation/simple_sensor_simulator/include/simple_sensor_simulator/vehicle_simulation/vehicle_model/sim_model_delay_steer_map_acc_geared.hpp

```diff
@@ -0,0 +1,338 @@
+};
+
+class AccelerationMap
+{
+public:
+  bool readAccelerationMapFromCSV(const std::string & csv_path)
+  {
+    CSVLoader csv(csv_path);
+    std::vector<std::vector<std::string>> table;
+    if (!csv.readCSV(table)) {
+      std::cerr << "[SimModelDelaySteerMapAccGeared]: failed to read acceleration map from "
+                << csv_path << std::endl;
+      return false;
+    }
+
+    vehicle_name_ = table[0][0];
+    vel_index_ = CSVLoader::getRowIndex(table);
+    acc_index_ = CSVLoader::getColumnIndex(table);
+    acceleration_map_ = CSVLoader::getMap(table);
+
+    std::cout << "[SimModelDelaySteerMapAccGeared]: success to read acceleration map from "
+              << csv_path << std::endl;
+    return true;
+  }
+
+  double getAcceleration(const double acc_des, const double vel) const
+  {
+    std::vector<double> interpolated_acc_vec;
+    const double clamped_vel = CSVLoader::clampValue(vel, vel_index_, "acc: vel");
+
+    // (throttle, vel, acc) map => (throttle, acc) map by fixing vel
+    for (const auto & acc_vec : acceleration_map_) {
+      interpolated_acc_vec.push_back(interpolation::lerp(vel_index_, acc_vec, clamped_vel));
+    }
+    // calculate throttle
+    // When the desired acceleration is smaller than the throttle area, return min acc
+    // When the desired acceleration is greater than the throttle area, return max acc
+    const double clamped_acc = CSVLoader::clampValue(acc_des, acc_index_, "acceleration: acc");
+    const double acc = interpolation::lerp(acc_index_, interpolated_acc_vec, clamped_acc);
+
+    return acc;
+  }
+  std::vector<std::vector<double>> acceleration_map_;
+
+private:
+  std::string vehicle_name_;
+  std::vector<double> vel_index_;
+  std::vector<double> acc_index_;
+};
+
+class SimModelDelaySteerMapAccGeared : public SimModelInterface
+{
+public:
+  /**
+     * @brief constructor
+     * @param [in] vx_lim velocity limit [m/s]
+     * @param [in] steer_lim steering limit [rad]
+     * @param [in] vx_rate_lim acceleration limit [m/ss]
+     * @param [in] steer_rate_lim steering angular velocity limit [rad/ss]
+     * @param [in] wheelbase vehicle wheelbase length [m]
+     * @param [in] dt delta time information to set input buffer for delay
+     * @param [in] acc_delay time delay for accel command [s]
+     * @param [in] acc_time_constant time constant for 1D model of accel dynamics
+     * @param [in] steer_delay time delay for steering command [s]
+     * @param [in] steer_time_constant time constant for 1D model of steering dynamics
+     * @param [in] path path to csv file for acceleration conversion map
+     */
+  SimModelDelaySteerMapAccGeared(
+    double vx_lim, double steer_lim, double vx_rate_lim, double steer_rate_lim, double wheelbase,
+    double dt, double acc_delay, double acc_time_constant, double steer_delay,
+    double steer_time_constant, std::string path);
+
+  /**
+     * @brief default destructor
+     */
+  ~SimModelDelaySteerMapAccGeared() = default;
```

Comment:
Please fix indent.

Reply:
fixed in https://github.com/tier4/scenario_simulator_v2/pull/1144/commits/1d8fdcf70587c5636d2b327722005e15ebbc42d4